src/cloudwatch-mcp-server/awslabs/cloudwatch_mcp_server/cloudwatch_metrics/tools.py

```diff
@@ -146,66 +146,106 @@ def register(self, mcp):
             target_datapoints: Target number of data points to return (default: 60). Controls the granularity of the returned data.
+            group_by_dimension: Dimension name to group by in Metrics Insights mode. Must be included in schema_dimension_keys.
+            schema_dimension_keys: List of dimension keys to include in the SCHEMA definition for Metrics Insights query.
+            limit: Maximum number of results to return in Metrics Insights mode (used with LIMIT clause).
+            sort_order: Sort order for results when using ORDER BY in Metrics Insights. Can be 'ASC', 'DESC', or None.
+            order_by_statistic: Statistic to use in the ORDER BY clause. Can be 'AVG', 'COUNT', 'MAX', 'MIN', 'SUM'. If not provided, uses the same statistic as the SELECT clause.
 
         Returns:
             GetMetricDataResponse: An object containing the metric data results
 
-        Example:
+        Example 1 (Standard GetMetricData):
             result = await get_metric_data(
                 ctx,
                 namespace="AWS/EC2",
                 metric_name="CPUUtilization",
+                start_time="2023-01-01T00:00:00Z",
                 dimensions=[
                     Dimension(name="InstanceId", value="i-1234567890abcdef0")
                 ],
-                start_time="2023-01-01T00:00:00Z",
-                # period will be auto-calculated based on time window
                 statistic="Average"
+                # Period will be auto-calculated based on time window and target_datapoints
             )
+            
+        Example 2 (Metrics Insights with group by):
+            result = await get_metric_data(
+                ctx,
+                namespace="AWS/EC2",
+                metric_name="CPUUtilization",
+                start_time="2023-01-01T00:00:00Z",
+                end_time="2023-01-02T00:00:00Z",
+                statistic="AVG",
+                schema_dimension_keys=["InstanceType"],
+                group_by_dimension="InstanceType"
+                # This will generate a query like: SELECT AVG("CPUUtilization") FROM SCHEMA("AWS/EC2", "InstanceType") GROUP BY "InstanceType"
+            )
+            
+        Example 3 (Metrics Insights with schema dimension keys):
+            result = await get_metric_data(
+                ctx,
+                namespace="AWS/EC2",
+                metric_name="CPUUtilization",
+                start_time="2023-01-01T00:00:00Z",
+                end_time="2023-01-02T00:00:00Z",
+                statistic="AVG",
+                schema_dimension_keys=["InstanceId", "InstanceType"],
+                group_by_dimension="InstanceId"
+                # This will generate a query like: SELECT AVG("CPUUtilization") FROM SCHEMA("AWS/EC2", "InstanceId", "InstanceType") GROUP BY "InstanceId"
+            )
+            
+        Example 4 (Metrics Insights with ORDER BY and LIMIT):
```

Comment:
nit: should we use a use-case driven description here? E.g.: `Find the top 5 EC2 instances with the highest CPU utilization`? AFAIK in MI syntax you can omit that.